server/series/api.py

```diff
@@ -0,0 +1,436 @@
+
+    try:
+        series = Series.objects.get(id=id) if id is not None else Series.objects.get(slug=slug)
+    except Series.DoesNotExist:
+        return 400, {"message": "Series does not exist"}
+
+    return 200, series
+
+
+@router.post("/", response={200: SeriesSchema, 400: Response, 401: Response})
+def create_series(
+    request: AuthenticatedHttpRequest, series_details: SeriesCreateSchema
+) -> tuple[int, Series | message_response]:
+    if not request.user.is_staff:
+        return 401, {"message": "Only Admins can create a series"}
+
+    series = Series(
+        start_date=series_details.start_date,
+        end_date=series_details.end_date,
+        name=series_details.name,
+        type=series_details.type,
+        category=series_details.category,
+        series_roster_max_players=series_details.series_roster_max_players,
+        event_min_players_male=series_details.event_min_players_male,
+        event_min_players_female=series_details.event_min_players_female,
+        event_max_players_male=series_details.event_max_players_male,
+        event_max_players_female=series_details.event_max_players_female,
+    )
+
+    if series_details.player_transfer_window_start_date:
+        series.player_transfer_window_start_date = series_details.player_transfer_window_start_date
+
+    if series_details.player_transfer_window_end_date:
+        series.player_transfer_window_end_date = series_details.player_transfer_window_end_date
+
+    if series_details.season_id:
+        try:
+            series.season = Season.objects.get(id=series_details.season_id)
+        except Season.DoesNotExist:
+            return 400, {"message": "Season does not exist"}
+
+    series.save()
+    return 200, series
+
+
+@router.put(
+    "/{series_slug}/register-team", response={200: SeriesSchema, 400: Response, 401: Response}
+)
+def add_team_series_registration(
+    request: AuthenticatedHttpRequest,
+    series_slug: str,
+    team_details: AddOrRemoveTeamSeriesRegistrationSchema,
+) -> tuple[int, Series | message_response]:
+    try:
+        series = Series.objects.get(slug=series_slug)
+        team = Team.objects.get(slug=team_details.team_slug)
+    except (Series.DoesNotExist, Team.DoesNotExist):
+        return 400, {"message": "Series/Team does not exist"}
+
+    if request.user not in team.admins.all():
+        return 401, {"message": "Only team admins can register a team to a series !"}
+
+    # Any team can register for the Club series
+    # School, College, State teams can register for the appropriate series only
+    if (
+        series.category != Series.Category.CLUB  # noqa: PLR1714
+        and series.category != team.category
+    ):
+        return 400, {"message": f"Only {series.category} teams can register for this series"}
+    else:
+        series.teams.add(team)
+
+    return 200, series
+
+
+@router.put(
+    "/{series_slug}/deregister-team", response={200: SeriesSchema, 400: Response, 401: Response}
```

Comment:
api can be `DELETE /{series_slug}/team/{team_slug}`.. so there is proper GET, POST and DELETE for this api and its self explanatory

Reply:
yeah i'll change